addons/project/models/project.py

```diff
@@ -1928,6 +1934,9 @@ def write(self, vals):
             raise UserError(_('You cannot archive recurring tasks. Please disable the recurrence first.'))
         if 'recurrence_id' in vals and vals.get('recurrence_id') and any(not task.active for task in self):
             raise UserError(_('Archived tasks cannot be recurring. Please unarchive the task first.'))
+        if 'child_ids' in vals and not self.ensure_one():
+            raise UserError(_("Sorry. a Task can only have one parent."))
+
```

Comment:
This creates a translation string for an error that is never occurring under normal circumstances, I don't think we usually add those.
Also, `ensure_one` is not returning a boolean, it's raising an exception when self contains more than one record.
What you could do is add a `self.ensure_one()` in your `if` a bit below, like this:
```py
        if 'child_ids' in vals:
            self.ensure_one()
            task = self.sudo()
            for follower in task.message_follower_ids:
                (task.child_ids - old_child_ids).message_subscribe(follower.partner_id.ids, follower.subtype_ids.ids)
```
This will raise an exception if you try to write on the `child_ids` field on multiple records at once, but since it shouldn't ever happen, I'm not sure it's even necessary, maybe to discuss during the final review.